htap.go

```diff
@@ -22,5 +25,41 @@ func (qcc *queryContextCache) init() *queryContextCache {
 func (qcc *queryContextCache) add(qces ...queryContextEntry) {
 	qcc.mutex.Lock()
 	defer qcc.mutex.Unlock()
-	qcc.entries = append(qcc.entries, qces...)
+	var newQcesBasedOnPriorities []queryContextEntry
+	for _, newQce := range qces {
+		logger.Debugf("adding query context: %v", newQce)
+		newQceProcessed := false
+		for existingQceIdx, existingQce := range qcc.entries {
+			if newQce.ID == existingQce.ID {
+				newQceProcessed = true
+				if newQce.Timestamp > existingQce.Timestamp {
+					if newQce.Priority == existingQce.Priority {
+						qcc.entries[existingQceIdx].Timestamp = newQce.Timestamp
+						qcc.entries[existingQceIdx].Context = newQce.Context
+					} else {
+						qcc.entries[existingQceIdx] = newQce
+					}
+				} else if newQce.Timestamp == existingQce.Timestamp {
+					if newQce.Priority != existingQce.Priority {
```

Comment:
could we join if conditions and don't have nested one?

Reply:
For me it is more readable, as this upper `else if` is connected to `if` also checking timestamp. So this level of indentation is about timestamps and the lower one is another is for priority.